### PR TITLE
Associate robot with plant

### LINKED ISSUE #743: A robot should be associated with a plant
**Describe the improvement you would like to see**
The robot model should be associated with the name of the asset it is located on.

- A field should be added to the robot model which gives the asset which the robot is located on.

**How will this change existing functionality?**
This will enable us to display only robots on the chosen asset.

**How will this improvement affect the current Threat Model?**
Not affected

### PR DESCRIPTION
**Pull request by OleDrange**
closes #743 
Connected to https://github.com/equinor/isar/pull/463
Waiting for #776 to merge before add database migrations

### CODE REVIEW

File: frontend/src/components/Pages/FrontPage/RobotCards/RobotStatusView.tsx [hunk 1]
```diff
@@ -40,10 +44,26 @@ export function RobotStatusSection({ refreshInterval }: RefreshProps) {
...
+
+    if (ActiveAsset() === '') {
+        robotDisplay = robots.map(function (robot) {
+            return <RobotStatusCard key={robot.id} robot={robot} />
+        })
+    } else {
+        robotDisplay = filteredRobots.map(function (robot) {
+            return <RobotStatusCard key={robot.id} robot={robot} />
+        })
+    }
```
Comment (GodVenn) [inline frontend/src/components/Pages/FrontPage/RobotCards/RobotStatusView.tsx:65]: Have you tested that the robots change when you change the asset?

**Review by GodVenn — COMMENTED** [review]
Should also add an update of robot on receiving IsarRobotInfo if the old robotAsset is different than the new one (like we currently are doing for the robot.Port and robot.Host (Line 178 - 188 in `MqttEventHandler.cs`)

File: backend/api/Database/Models/Robot.cs [hunk 2]
```diff
@@ -26,6 +26,8 @@ public class Robot
         [MaxLength(200)]
         public string SerialNumber { get; set; }
 
+        public string RobotAsset { get; set; }
```
Comment (GodVenn) [inline backend/api/Database/Models/Robot.cs:29]:
```suggestion
        public string CurrentAsset { get; set; }
```
Don't include the name of the parent object as part of the child parameter

File: frontend/src/components/Pages/FrontPage/RobotCards/RobotStatusView.tsx (hunk 1 above)
Comment (OleDrange) [reply frontend/src/components/Pages/FrontPage/RobotCards/RobotStatusView.tsx:65]: Yes

File: backend/api/Database/Models/Robot.cs (hunk 2 above)
Comment (OleDrange) [reply backend/api/Database/Models/Robot.cs]: done

**Comment by OleDrange** [comment]
> Should also add an update of robot on receiving IsarRobotInfo if the old robotAsset is different than the new one (like we currently are doing for the robot.Port and robot.Host (Line 178 - 188 in `MqttEventHandler.cs`)

done

File: frontend/src/models/Robot.ts [hunk 3]
```diff
@@ -14,6 +14,7 @@ export interface Robot {
     name?: string
     model: RobotModel
     serialNumber?: string
+    currentAsset?: string
```
Comment (GodVenn) [inline frontend/src/models/Robot.ts:17]:
```suggestion
    currentAsset: string
```
If its not nullable in the backend it shouldn't be nullable here.

**Review by GodVenn — COMMENTED** [review]
I think the database update will fail due to adding a required column,but we can look at it together to fix that

File: frontend/src/models/Robot.ts (hunk 3 above)
Comment (OleDrange) [reply frontend/src/models/Robot.ts]: done

**Review by GodVenn — APPROVED** [review]
LGTM

**Pull request merged**